### PR TITLE
connman: improved configuration

### PR DESCRIPTION
**Pull request by ericsagnes**
connman:
- blacklisting NixOS containers by default
- added option for extra configurations

details are in #9628

cc package maintainer @matejc

### CODE REVIEW

File: nixos/modules/services/networking/connman.nix [hunk 1]
```diff
@@ -22,6 +27,23 @@ in {
...
+        type = types.lines;
+        default = ''
+        '';
+        description = ''
+          Configuration lines appended to the generated connman configuration file.
+        '';
+      };
+
+      networkInterfaceBlacklist = mkOption {
+        type = types.str;
```
Comment (copumpkin) [inline nixos/modules/services/networking/connman.nix]: Couldn't this easily be a list of strings on the Nix side? Might be a little easier to do unusual things with.

**Comment by ericsagnes** [comment]
@copumpkin Thanks for the advice!
Indeed a list of string here adds more flexibility, I updated the PR.

**Comment by jagajaga** [comment]
"+1"

**Comment by matejc** [comment]
@ericsagnes ok this looks great, let me try it .. if all checks out I will merge

**Comment by matejc** [comment]
@ericsagnes thanks!

**Pull request merged**